### PR TITLE
PIL-647  updates to Group status & Reg conf page

### PR DESCRIPTION
**Pull request by javed-iqbal1**

### CODE REVIEW

File: app/controllers/RegistrationConfirmationController.scala [hunk 1]
```diff
@@ -38,6 +41,12 @@ class RegistrationConfirmationController @Inject() (
     with I18nSupport {
 
   def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
-    Ok(view(request.session.get(Pillar2SessionKeys.plrId).getOrElse("N/A")))
+    val currentDate = HtmlFormat.escape(dateHelper.formatDateGDS(java.time.LocalDate.now))
+    val mneDomestic = request.session.data.get(Pillar2SessionKeys.updateMneOrDomestic) match {
+      case Some("ukAndOther") => "Domestic Top-up Tax and Multinational Top-up Tax"
```
Comment (AKKUM) [inline app/controllers/RegistrationConfirmationController.scala:46]: Better to put the content in Constant class.

File: conf/messages.en [hunk 2]
```diff
@@ -603,10 +613,17 @@ site.confirm.send = Confirm and send
...
-registrationConfirmation.p1 = You have set up an account for reporting and managing Pillar 2 top-up taxes.
-registrationConfirmation.p2 = You can save this ID number by making a note or taking a screenshot.
+registrationConfirmation.p1 = Your group has registered to report for {0}.
+registrationConfirmation.p2 = You will not be emailed a confirmation of this registration.
+registrationConfirmation.p3 = You can save a record of this registration completion as a PDF (depending on your browser it will open in a new window or tab, or save to your download folder).
+registrationConfirmation.link = Save a record of this registration completion
+registrationConfirmation.p4 = You must not disclose your Pillar 2 top-up taxes ID or registration date outside of agreed communications.
+registrationConfirmation.next = What happens next
+registrationConfirmation.now = You can now
+registrationConfirmation.link-home =  report and manage your Pillar 2 top-up taxes.
```
Comment (jassalrichy) [inline conf/messages.en:626]: Full stop should be outside of link text.
See: [link-home](https://design-system.service.gov.uk/styles/links/)

**Comment by aashik-varahan** [comment]
@AKKUM @jassalrichy - Resolved the suggestions. Please review and approve.

File: app/cache/SessionData.scala [hunk 3]
```diff
@@ -38,4 +38,8 @@ class SessionData {
     request.session +
       (Pillar2SessionKeys.registeringNfmForThisGroup -> value)
 
+  def updateMneOrDomestic(value: String)(implicit request: Request[AnyContent]): Session =
+    request.session +
+      (Pillar2SessionKeys.updateMneOrDomestic -> value)
+
```
Comment (jassalrichy) [inline app/cache/SessionData.scala:44]: Can we add a unit test for this please.

File: test/controllers/RegistrationConfirmationControllerSpec.scala [hunk 4]
```diff
@@ -33,12 +35,16 @@ class RegistrationConfirmationControllerSpec extends SpecBase {
...
+        val currentDate = HtmlFormat.escape(dateHelper.formatDateGDS(java.time.LocalDate.now))
+        val view        = application.injector.instanceOf[RegistrationConfirmationView]
 
         status(result) mustEqual OK
-        contentAsString(result) mustEqual view("N/A")(request, appConfig(application), messages(application)).toString
+        contentAsString(result) mustEqual view("N/A", currentDate.toString(), "Domestic Top-up Tax")(
+          request,
+          appConfig(application),
+          messages(application)
+        ).toString
```
Comment (jassalrichy) [inline test/controllers/RegistrationConfirmationControllerSpec.scala:47]: Need a test for when content is equal to FrontendAppConfig.registrationControllerMne value

File: app/views/subscriptionview/MneOrDomesticView.scala.html [hunk 5]
```diff
@@ -35,10 +35,20 @@
...
+        <p class="govuk-body">@messages("mneOrDomestic.p4")</p>
+        <p class="govuk-body">@messages("mneOrDomestic.p5")</p>
         @govukRadios(
             RadiosViewModel(
                 field  = form("value"),
-                legend = LegendViewModel(messages("mneOrDomestic.heading")).asPageHeading(Large),
+                legend = LegendViewModel(messages("mneOrDomestic.group-operate")).asPageHeading(Medium),
                 items  = MneOrDomestic.options
             )
         )
```
Comment (jassalrichy) [inline app/views/subscriptionview/MneOrDomesticView.scala.html:54]: I've noticed that we don't have any unit tests for the views. Will discuss with @AKKUM  if this should be added moving forward.

**Review by jassalrichy — APPROVED** [review]

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**